ras_party/controllers/controller.py

```diff
@@ -531,8 +526,86 @@ def post_case_event(case_id, party_id, category="Default category message", desc
     return response.json()
 
 
-def notify(email, template_id, url, party_id):
-    # TODO: Comments and explanation
+def _query_respondent_by_party_uuid(party_uuid):
+    """
+    Query to return respondent based on party uuid
+    :param party_uuid: the pary uuid
+    :return: respondent
+    """
+    log.debug('Querying respondents with party_uuid {}'.format(party_uuid))
+
+    with db_session() as tran:
+        return tran.query(Respondent).filter(Respondent.party_uuid == party_uuid).first()
+
+
+def resend_verification_email(party_uuid):
+    """
+    Check and resend an email verification email
+    :param party_uuid: the party uuid
+    :return: make_response
+    """
+    log.debug('attempting to resend verification_email with party_uuid {}'.format(party_uuid))
+
+    respondent = _query_respondent_by_party_uuid(party_uuid)
+
+    if not respondent:
```

Comment:
A minor point, may be worth considering raising an exception for the unhappy path flow which will log the message and return a formatted error response.